Sources/CoreMetrics/Metrics.swift

```diff
@@ -679,6 +768,54 @@ internal final class AccumulatingRoundingFloatingPointCounter: FloatingPointCoun
     }
 }
 
+/// Wraps a RecorderHandler, adding support for incrementing values by storing an accumulated  value and recording increments to the underlying CounterHandler after crossing integer boundaries.
+internal final class AccumulatingCounter: GaugeHandler {
+    private let recorderHandler: RecorderHandler
+    // FIXME: use atomics when available
```

Comment:
Oh I guess we could follow up on this separately very soon, swift-atomics are doing well nowadays.